plugins/modules/net_tools/dnsimple_info.py

```diff
@@ -252,84 +250,85 @@ def build_url(account, key, is_sandbox):
-        data = response.json()["data"]
-        pages = response.json()["pagination"]["total_pages"]
-        if int(pages) > 1:
-            for page in range(1, pages):
-                page = page + 1
-                request_object.url = base_url + '&page=' + str(page)
-                new_results = Session().send(request_object)
-                data = data + new_results.json()["data"]
-        return(data)
-    else:
+    if 'pagination' not in response.json():
         module.fail_json('API Call failed, check ID, key and sandbox values')
 
+    data = response.json()['data']
+    total_pages = int(response.json()['pagination']['total_pages'])
+    page = 1
+    while page < total_pages:
+        page = page + 1
+        request_object.url = '{url}&page={page}'.format(url=base_url, page=page)
+        new_results = Session().send(request_object)
+        data = data + new_results.json()['data']
+
+    return data
+
 
 def record_info(dnsimple_mod, req_obj):
-    req_obj.url, req_obj.method = req_obj.url + '/zones/' + dnsimple_mod.params["name"] + '/records?name=' + dnsimple_mod.params["record"], 'GET'
+    req_obj.url, req_obj.method = '{url}/zones/{name}/records/?name={record}'.format(
+        url=req_obj.url,
+        name=dnsimple_mod.params['name'],
+        record=dnsimple_mod.params['record']), 'GET'
     return iterate_data(dnsimple_mod, req_obj)
 
 
 def domain_info(dnsimple_mod, req_obj):
-    req_obj.url, req_obj.method = req_obj.url + '/zones/' + dnsimple_mod.params["name"] + '/records?per_page=100', 'GET'
+    req_obj.url, req_obj.method = '{url}/zones/{name}/records?per_page=100'.format(url=req_obj.url, name=dnsimple_mod.params['name']), 'GET'
     return iterate_data(dnsimple_mod, req_obj)
 
 
 def account_info(dnsimple_mod, req_obj):
-    req_obj.url, req_obj.method = req_obj.url + '/zones/?per_page=100', 'GET'
+    req_obj.url, req_obj.method = '{url}/zones/?per_page=100'.format(url=req_obj.url), 'GET'
     return iterate_data(dnsimple_mod, req_obj)
 
 
 def main():
-    # define available arguments/parameters a user can pass to the module
-    fields = {
-        "account_id": {"required": True, "type": "str"},
-        "api_key": {"required": True, "type": "str", "no_log": True},
-        "name": {"required": False, "type": "str"},
-        "record": {"required": False, "type": "str"},
-        "sandbox": {"required": False, "type": "bool", "default": False}
-    }
-
     result = {
         'changed': False
     }
 
     module = AnsibleModule(
-        argument_spec=fields,
-        supports_check_mode=True
+        argument_spec={
+            'account_id': {'required': True, 'type': 'str'},
+            'api_key': {'required': True, 'type': 'str', 'no_log': True},
+            'name': {'type': 'str'},
+            'record': {'type': 'str'},
+            'sandbox': {'type': 'bool', 'default': False}
+        },
+        supports_check_mode=True,
+        required_by={
+            'record': 'name',
```

Comment:
I think this is a breaking change :D How about deprecating usage of `record` without `name` first so that we can add this eventually? (And until then, comment this out, so that whoever processes the deprecation doesn't forget to use this solution?)

Reply:
More like a bugfix. Passing record without name, in the old code, will fall in the `if-else` where all domains are returned, which is clearly not the desired outcome for this.

What about submitting a bugfix for that first, then these improvements?

Reply:
Well, this will still potentially break playbooks that worked before (even inadvertedly). That's why we should use a deprecation here to make the playbook authors aware that they were doing crap and they have to fix it, before we eventually disallow it :)